### ISSUE TITLE
Added new colors to $theme-colors but related classes are not created for text and background colors

### ISSUE THREAD

**Issue by bkilinc** [twbs/bootstrap#35297]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

I added custom colors to $theme-colors map, btn-, link- colors are generated perfectly but text- and bg- colors are not generated. I guess it should behave same. 

my custom.scss file is as follows.
```scss
@import "~bootstrap/scss/functions";
@import "~bootstrap/scss/variables";

$custom-colors: (
  "primary-light": lighten(map-get($theme-colors,"primary"), 20),
  "secondary-light": lighten(map-get($theme-colors,"secondary"), 20),
  "success-light": lighten(map-get($theme-colors,"success"), 20),
  "info-light": lighten(map-get($theme-colors,"info"), 20),
  "warning-light": lighten(map-get($theme-colors,"warning"), 20),
  "danger-light": lighten(map-get($theme-colors,"danger"), 20)
);

// Merge the maps
$theme-colors: map-merge($theme-colors, $custom-colors);
 ```
as a result  ;
* btn btn-primary-light, link-primary-light classes are generated correctly
* bg-primary-light, text-primary-light clases are not generated as expected.

### Reduced test cases

include file at top;
```scss
@import "~bootstrap/scss/functions";
@import "~bootstrap/scss/variables";

$custom-colors: (
  "primary-light": lighten(map-get($theme-colors,"primary"), 20),
  "secondary-light": lighten(map-get($theme-colors,"secondary"), 20),
  "success-light": lighten(map-get($theme-colors,"success"), 20),
  "info-light": lighten(map-get($theme-colors,"info"), 20),
  "warning-light": lighten(map-get($theme-colors,"warning"), 20),
  "danger-light": lighten(map-get($theme-colors,"danger"), 20)
);

// Merge the maps
$theme-colors: map-merge($theme-colors, $custom-colors);
 ```

bg-primary-light, text-primary-light should be generated

### What operating system(s) are you seeing the problem on?

Linux

### What browser(s) are you seeing the problem on?

Firefox

### What version of Bootstrap are you using?

v5.1.3

**Comment by MisatoTremor** [comment]
The maps that are used for these are built in `_variables.scss`, but you can rebuild them.

Instead of just 
```scss
$theme-colors: map-merge($theme-colors, $custom-colors);
```
do
```scss
$theme-colors: map-merge($theme-colors, $custom-colors);
$theme-colors-rgb: map-loop($theme-colors, to-rgb, "$value");
$utilities-colors: map-merge($utilities-colors, $theme-colors-rgb);
$utilities-text: map-merge($utilities-text, $utilities-colors);
$utilities-text-colors: map-loop($utilities-text, rgba-css-var, "$key", "text");
$utilities-bg: map-merge($utilities-bg, $utilities-colors);
$utilities-bg-colors: map-loop($utilities-bg, rgba-css-var, "$key", "bg");
```

**Comment by bkilinc** [comment]
This works. But, it isn't beautiful.

**Comment by mdo** [comment]
Duplicate of #34756. #34942 will be shipped in v5.2.0 to address it.

**Comment by snarea** [comment]
I solved the problem.
Related information exists in [ helpers/color-bg.scss ].
I worked with the following sources.*

`
$theme-colors: map-merge($theme-colors, $custom-theme-colors);
$theme-colors-rgb: map-loop($theme-colors, to-rgb, "$value");
$utilities-colors: map-merge($utilities-colors, $theme-colors-rgb);
$utilities-text: map-merge($utilities-text, $utilities-colors);
$utilities-text-colors: map-loop($utilities-text, rgba-css-var, "$key", "text");
$utilities-bg: map-merge($utilities-bg, $utilities-colors);
$utilities-bg-colors: map-loop($utilities-bg, rgba-css-var, "$key", "bg");

@each $color, $value in $theme-colors {
  $color-rgb: to-rgb($value);
  .text-bg-#{$color} {
    color: color-contrast($value) !important;
    background-color: RGBA($color-rgb, var(--#{$prefix}bg-opacity, 1)) !important;
  }
}
@each $color, $value in $theme-colors {
  $color-rgb: to-rgb($value);
  .link-#{$color} {
    color: color-contrast($value) !important;
  }
}

`
